src/java.base/share/classes/java/util/FormatProcessor.java

```diff
@@ -0,0 +1,284 @@
+     * specifications found in the fragments and values in the
+     * supplied {@link StringTemplate} object. This method constructs a
+     * format string from the fragments, gathers up the values and
+     * evaluates the expression
+     * {@code new Formatter(locale).format(format, values).toString()}.
+     * <p>
+     * If an embedded expression is not immediately preceded by a
+     * specifier then a {@code %s} is inserted in the format.
+     *
+     * @param stringTemplate  a {@link StringTemplate} instance
+     *
+     * @return constructed {@link String}
+
+     * @throws  IllegalFormatException
+     *          If a format specifier contains an illegal syntax, a format
+     *          specifier that is incompatible with the given arguments,
+     *          a specifier not followed immediately by an embedded expression or
+     *          other illegal conditions. For specification of all possible
+     *          formatting errors, see the
+     *          <a href="../util/Formatter.html#detail">details</a>
+     *          section of the formatter class specification.
+     *
+     * @see java.util.Formatter
+     */
+    @Override
+    public final String process(StringTemplate stringTemplate) {
+        Objects.requireNonNull(stringTemplate);
+        String format = stringTemplateFormat(stringTemplate.fragments());
+        Object[] values = stringTemplate.values().toArray(new Object[0]);
+
+        return new Formatter(locale).format(format, values).toString();
+    }
+
+    /**
+     * Constructs a {@link MethodHandle} that when supplied with the values from
+     * a {@link StringTemplate} will produce a result equivalent to that provided by
+     * {@link FormatProcessor#process(StringTemplate)}. This {@link MethodHandle}
+     * is used by {@link FormatProcessor#FMT} and the ilk to perform a more
+     * specialized composition of a result. This is specialization is done by
+     * prescanning the fragments and value types of a {@link StringTemplate}.
+     * <p>
+     * Process template expressions can be specialized  when the processor is
+     * of type {@link ProcessorLinkage} and fetched from a static constant as is
+     * {@link FormatProcessor#FMT} ({@code static final FormatProcessor}).
+     * <p>
+     * Other {@link FormatProcessor} can be specialized if stored as static final.
+     * For example:
+     * {@snippet :
+     * FormatProcessor THAI_FMT = FormatProcessor.create(Locale.forLanguageTag("th-TH-u-nu-thai"));
+     * }
+     * {@code THAI_FMT} will now produce specialized {@link MethodHandle MethodHandles} by way
+     * of {@link FormatProcessor#linkage(List, MethodType)}.
+     *
+     * See {@link FormatProcessor#process(StringTemplate)} for more information.
+     *
+     * @throws  IllegalFormatException
+     *          If a format specifier contains an illegal syntax, a format
+     *          specifier that is incompatible with the given arguments,
```

Comment:
```suggestion
     *          specifier is incompatible with the given arguments,
```

Reply:
Existing statement is consistent with those in Formatter. Reads more correctly as is.